### ISSUE TITLE
Fetched data discrepancies

### ISSUE THREAD

**Issue by almalh** [ranaroussi/yfinance#218]
I've noticed that the fetched data is sometimes different from the data shown on Yahoo. For example, for CAE.TO, the Capital Expenditures values I get using Ticker.cashflow (-454 500k for 2019) are different from the ones at https://finance.yahoo.com/quote/CAE.TO/cash-flow?p=CAE.TO (-338 400k for the same year). Is there any explanation?

**Comment by cwtan501** [comment]
Yes, I am facing the same issue as well. Take Microsoft (MSFT) as example, the prices retrieved using the API are not tallied with those shown on Yahoo Finance. Truly appreciate someone could provide insights about this issue?

**Comment by aiden1yoon** [comment]
Can confirm the issue as well. Various data points are different from what's shown in Yahoo Finance website. However, the issue wasn't prevalent as the information from AAPL was ok but MSFT was wrong on the same data point.

**Comment by seanhsiung94** [comment]
I noticed a pattern that the closing price will be different before a recent dividend payment date. 

e.g. (T) had dividend payout on April 8th, 2020. The prices fetched after date is consistent with the website; while any date before the payment date is off. T has a closing price of $30.08 on April 7th, 2020, but the data fetched from .history api returns $29.56. The difference is exactly the dividend/share: $0.52.
